scripts/plotParticles.py

```diff
@@ -1,64 +1,86 @@
-    if tracerfile != 'none':
-      tfile = Dataset(tracerfile,'r')
-      X = tfile.variables['x']
-      Y = tfile.variables['y']
-      P = tfile.variables['P']
-      plt.contourf(np.squeeze(X),np.squeeze(Y),np.squeeze(P))
+    if(recordedvar is not 'none'):
+        rMax = np.max(pfile.variables[recordedvar])
+        record = pfile.variables[recordedvar]
+
+    if backgroundfield is not 'none':
+        bfile = Dataset(backgroundfield, 'r')
+        bX = bfile.variables['x']
+        bY = bfile.variables['y']
+        bT = bfile.variables['time_counter']
+        # Find the variable that exists across at least two spatial and one time dimension
+
+        def checkShape(var):
+            if len(np.shape(var)) > 3:
+                return True
+        for v in bfile.variables:
+            if checkShape(bfile.variables[v]):
+                bVar = bfile.variables[v]
+        bMax = np.max(bVar)
 
     if mode == '3d':
-      ax = fig.gca(projection='3d')
-      for p in range(len(lon)):
-        ax.plot(lon[p,:],lat[p,:],z[p,:],'.-')
-      ax.set_xlabel('Longitude')
-      ax.set_ylabel('Latitude')
-      ax.set_zlabel('Depth')
+        fig = plt.figure(1)
+        ax = fig.gca(projection='3d')
+        for p in range(len(lon)):
+            ax.plot(lon[p, :], lat[p, :], z[p, :], '.-')
+        ax.set_xlabel('Longitude')
+        ax.set_ylabel('Latitude')
+        ax.set_zlabel('Depth')
     elif mode == '2d':
-      plt.plot(np.transpose(lon),np.transpose(lat),'.-')
-      plt.xlabel('Longitude')
-      plt.ylabel('Latitude')    
-    elif mode =='movie2d':
+        if backgroundfield is not 'none':
+            plt.contourf(np.squeeze(bX), np.squeeze(bY), np.squeeze(bVar), zorder=-1, vmin=0, vmax=bMax)
+        plt.plot(np.transpose(lon), np.transpose(lat), '.-')
+        plt.xlabel('Longitude')
+        plt.ylabel('Latitude')
+    elif mode == 'movie2d':
+
+        fig = plt.figure(1)
+        ax = plt.axes(xlim=(np.amin(lon), np.amax(lon)), ylim=(np.amin(lat), np.amax(lat)))
+        ax.scatter(lon[:, 0], lat[:, 0], s=60, c='black')
 
-      line, = ax.plot(lon[:,0], lat[:,0],'ow')
-      if tracerfile == 'none': # need to set ax limits
-        plt.axis((np.amin(lon),np.amax(lon),np.amin(lat),np.amax(lat)))
-        
-      def animate(i):
-          line.set_xdata(lon[:,i])
-          line.set_ydata(lat[:,i])
-          return line,
+        def animate(i):
+            ax.cla()
+            if recordedvar is not 'none':
+                scat = ax.scatter(lon[:, i], lat[:, i], s=60, c=record[:, i]/rMax)
+            else:
+                scat = ax.scatter(lon[:, i], lat[:, i], s=60, c='white', edgecolors='black')
+            if backgroundfield is not 'none':
+                field_time = np.argmax(bT > time[0, i]) - 1
+                plt.contourf(bX[:], bY[:], bVar[field_time, 0, :, :], zorder=-1, vmin=0, vmax=bMax,
+                             levels=np.linspace(0, bMax, 10))
+            return scat,
```

Comment:
A lot of new code has landed with PR #56 and PR #59  doing prettyprinting of titles and axes etc; Again, is there some way to hook into that code in `parcels.py`?